test/fuzz/sql_fuzzer/sql_fuzzer.cc

```diff
@@ -0,0 +1,47 @@
+#include <string>
+#include <iostream>
+
+#include <libprotobuf-mutator/port/protobuf.h>
+#include <libprotobuf-mutator/src/libfuzzer/libfuzzer_macro.h>
+#include "sql_query.pb.h"
+#include "sql_query_proto_to_string.h"
+
+#include "fiber.h"
+#include "memory.h"
+#include "coll/coll.h"
+#include "box/box.h"
+#include "box/memtx_tx.h"
+#include "box/module_cache.h"
+#include "box/sql.h"
+
+#define SUCCESS 0
+
+DEFINE_PROTO_FUZZER(const sql_query::SQLQuery &query) {
+	std::string query_str = sql_fuzzer::SQLQueryToString(query);
+
+	if (::getenv("LPM_DUMP_NATIVE_INPUT") && query_str.size() != 0) {
+		std::cout << "_________________________" << std::endl;
+		std::cout << query_str << std::endl;
+		std::cout << "-------------------------" << std::endl;
+	}
+
+	if (query_str.length() == 0)
+		return;
+
+	memory_init();
+	fiber_init(fiber_cxx_invoke);
+	coll_init();
+	box_init();
+	sql_init();
+
+	struct Expr *expr = sql_expr_compile(sql_get(), query_str.data(),
```

Comment:
We decided to not use `sql_stmt_compile()` because it was causing errors.
```
AddressSanitizer:DEADLYSIGNAL
=================================================================
==71826==ERROR: AddressSanitizer: SEGV on unknown address 0x000000000010 (pc 0x561619fa375e bp 0x7ffe92a81100 sp 0x7ffe92a80ff0 T0)
==71826==The signal is caused by a READ memory access.
==71826==Hint: address points to the zero page.
    #0 0x561619fa375e in mh_i32ptr_find /home/klauwier/projects/tarantool/src/lib/salad/mhash.h:202:17
    #1 0x561619fa35ab in space_by_id /home/klauwier/projects/tarantool/src/box/space_cache.c:57:19
    #2 0x561619af3202 in vdbe_emit_halt_with_presence_test /home/klauwier/projects/tarantool/src/box/sql/build.c:3559:50
    #3 0x561619af2dcd in sqlEndTable /home/klauwier/projects/tarantool/src/box/sql/build.c:1478:6
    #4 0x56161a09543e in yy_reduce /home/klauwier/projects/tarantool/src/box/sql/parse.y:214:3
    #5 0x56161a0928ab in sqlParser /home/klauwier/projects/tarantool/build/src/box/sql/parse.c:4683:7
    #6 0x561619c2f966 in sqlRunParser /home/klauwier/projects/tarantool/src/box/sql/tokenize.c:546:4
    #7 0x561619bbdcf3 in sql_stmt_compile /home/klauwier/projects/tarantool/src/box/sql/prepare.c:81:4```